internal/tui/model.go

```diff
@@ -347,16 +350,41 @@ func (m Model) processUpdate(msg *progrock.StatusUpdate) (tea.Model, tea.Cmd) {
 	return m, tea.Batch(cmds...)
 }
 
+func (m Model) addToFirstGroup(id string) {
```

Comment:
A side fix; I recently changed it to include duplicates vertices, but I think it was ultimately just confusing DX, so now it just picks the first pipeline path that the vertex is seen in, the same way the Cloud telemetry flow works.